solutions/autoscaling/README.md

```diff
@@ -0,0 +1,127 @@
+# autoscale
+
+This is a demo of using avn CLI and prometheus endpoint to upsize an Aiven service when system resource reaches a defined threshold.  This demo would take around 10 - 15 minutes to be completed.
+
+### *This is NOT for production use, safe guards should be placed for any automation on changing service plans.*
+
+## Requirement
+
+- `avn` 
+- `curl`
+- `jq`
+- `psql`
+
+## Steps
+
+- Update the following values in `autoscale.env`
+
+| Tables                 | Description                 |
+| ---------------------- |:---------------------------:|
+| PROJECT                | project name                |
+| SERVICE                | service name                |
+| AUTOSCALE_THRESHOLD    | threshold to upsize         |  
+| AUTOSCALE_INTERVAL     | monito interval in seconds  |
+| AUTOSCALE_PLAN_TYPE    | service type                |
+| AUTOSCALE_CURRENT_PLAN | plan size to start with     |
+| AUTOSCALE_NEXT_PLAN    | plan size to upgrade to     |
+| CPU_IDLE_MAX           | percentage of cpu idle      |
+| PROMETHEUS_NAME        | promeheus endpoint name     |
+| PROMETHEUS_USER        | promeheus endpoint user     |
+| PROMETHEUS_PASS        | promeheus endpoint password |
+
+
+- Run `./autoscale.sh setup`, this would create the service and the prometheus integration endpoint that are needed for this demo.  It takes around 3 minutes to create a new service.
+```
+./autoscale.sh setup
+```
+
+- Run `./autoscale.sh demo`, this starts to check cpu idle percentage every 10 seconds, when it is less than `60%` which means overall cpu usage is over `40%` it will keep track of the duration, if this continues for 60 seconds, it will trigger `avn` CLI to upsize the plan to `startup-8`.  Leave this running in a terminal, this does not exit until `CTRL + C`, open another terminal to run `./autoscale.sh pgload`
+```
+./autoscale.sh demo
+```
+
+- This step runs an expensive PostgreSQL query to spike up CPU load that may take a few minutes to complete, it is expected that you will not get the sum response in over 1 minute.  Go to Aiven Console and open the service `Metrics` view to monitor the change in CPU usage, click on the `Nodes` icon to see node replacement during the plan upgrade, it is expected in `./autoscale.sh demo` terminal you will get `Unable to get metrics...` during the new node takes over.
```

Comment:
Add the line 
"Open a new terminal to monitor spike in CPU load when running the below command"